### ISSUE TITLE
Lint Reference Usage without namespace

### ISSUE THREAD

**Issue by Sysix** [stylelint-scss/stylelint-scss#595]
Hello,

it would really help when there is a rule for following use case:

```scss
// file a.scss:
@import 'b';
@import 'c';
```

```scss
// file b.scss:
$var1: #000;
```

```scss
// file c.scss:
body {
  color: $var1; <-- should warn
}
```

The solution should be:

```scss
// file c.scss:
@use 'b';

body {
  color: b.$var1;
}
```

This also should check for `@mixins` at other namespaceable keywords.

**Comment by kristerkari** [comment]
Thanks for the suggestion @Sysix ! 

I think that the problem with your suggestions is that the styles that you are referring to are separated to three files. 

The linter in this case would run against the pre-compiled Sass files, so it does not really have knowledge of what you have inside the other files, which means that it can't detect the thing that you want to warn about.

**Comment by Sysix** [comment]
hey @kristerkari 

Im just wondering how then the linter detect when you use `darken()` without `math::colors`.
The logic behind that should be also capable to detect custom functions / mixins / variables.

Or I'm missing something? What are the key differences?

**Comment by kristerkari** [comment]
> Im just wondering how then the linter detect when you use `darken()` without `sass::colors`. The logic behind that should be also capable to detect custom functions / mixins / variables.
> 
> Or I'm missing something? What are the key differences?

@Sysix That's good question. I'll explain what's the difference.

When you use Sass in your project, it takes care of compiling your SCSS styles to CSS. It also means that the compiler (Sass) has to take into account all the imports that you have in your files.

The key difference with the linter (stylelint) is that it does not compile your styles. What it does is to go file by file and parse the contents to something that we can analyze. That also means that when there is an import, the linter does not have access to the file that gets imported.

Detecting the `darken()` function is done by looking at the current file to see if the function gets imported from `sass::colors` or used as a global function. No compiling of the styles needed.
